src/snapshot_tests/test_case.rs

```diff
@@ -224,70 +234,126 @@ pub struct TestInput {
+    const COLOR_VARIANTS: [RGBColor; 17] = [
         // RED, input_0
-        (255, 0, 0),
+        RGBColor(255, 0, 0),
         // BLUE, input_1
-        (0, 0, 255),
+        RGBColor(0, 0, 255),
         // GREEN, input_2
-        (0, 255, 0),
+        RGBColor(0, 255, 0),
         // YELLOW, input_3
-        (255, 255, 0),
+        RGBColor(255, 255, 0),
         // MAGENTA, input_4
-        (255, 0, 255),
+        RGBColor(255, 0, 255),
         // CYAN, input_5
-        (0, 255, 255),
+        RGBColor(0, 255, 255),
         // ORANGE, input_6
-        (255, 165, 0),
+        RGBColor(255, 165, 0),
         // WHITE, input_7
-        (255, 255, 255),
+        RGBColor(255, 255, 255),
         // GRAY, input_8
-        (128, 128, 128),
+        RGBColor(128, 128, 128),
         // LIGHT_RED, input_9
-        (255, 128, 128),
+        RGBColor(255, 128, 128),
         // LIGHT_BLUE, input_10
-        (128, 128, 255),
+        RGBColor(128, 128, 255),
         // LIGHT_GREEN, input_11
-        (128, 255, 128),
+        RGBColor(128, 255, 128),
         // PINK, input_12
-        (255, 192, 203),
+        RGBColor(255, 192, 203),
         // PURPLE, input_13
-        (128, 0, 128),
+        RGBColor(128, 0, 128),
         // BROWN, input_14
-        (165, 42, 42),
+        RGBColor(165, 42, 42),
         // YELLOW_GREEN, input_15
-        (154, 205, 50),
+        RGBColor(154, 205, 50),
         // LIGHT_YELLOW, input_16
-        (255, 255, 224),
+        RGBColor(255, 255, 224),
     ];
 
     pub fn new(index: usize) -> Self {
-        Self::new_with_resolution(
+        Self::new_patterned(index, TestInputPattern::Plain)
+    }
+
+    pub fn new_with_resolution(index: usize, resolution: Resolution) -> Self {
+        Self::new_patterned_with_resolution(index, TestInputPattern::Plain, resolution)
+    }
+
+    pub fn new_patterned(index: usize, pattern: TestInputPattern) -> Self {
+        Self::new_patterned_with_resolution(
             index,
+            pattern,
             Resolution {
                 width: 640,
                 height: 360,
             },
         )
     }
 
-    pub fn new_with_resolution(index: usize, resolution: Resolution) -> Self {
-        if index >= Self::COLOR_VARIANTS.len() {
-            panic!("Reached input amount limit: {}", Self::COLOR_VARIANTS.len())
-        }
-
+    pub fn new_patterned_with_resolution(
+        index: usize,
+        pattern: TestInputPattern,
+        resolution: Resolution,
+    ) -> Self {
         let color = Self::COLOR_VARIANTS[index];
-        let r = color.0 as f32;
-        let g = color.1 as f32;
-        let b = color.2 as f32;
+        let mut y_plane = vec![0; resolution.width * resolution.height];
+        let mut u_plane = vec![0; (resolution.width * resolution.height) / 4];
+        let mut v_plane = vec![0; (resolution.width * resolution.height) / 4];
+
+        let rgb_color = |x: usize, y: usize| match pattern {
```

Comment:
I would suggest following approach:
- always add border, lets say 10 pixels
- divide input into grid of squares of constant size, for our tests probably 50 px border is fine. Half of the squares would be in normal color, and the other half in the alternative one. 
  - It's up to you, but I don't think you need to take into account border width when calculating grid
- generate alternative colors instead using black. It could be done by adding or subtracting sth from regular color's RGB values. This color does not need to be very distinct from original, we just need it to be visible where they are next to each other.